Sources/XMTPiOS/SigningKey.swift

```diff
@@ -29,6 +38,18 @@ public protocol SigningKey {
 }
 
 extension SigningKey {
+	public var isSmartContractWallet: Bool {
+		return false
+	}
+	
+	public var chainId: UInt64 {
+		return "eip155:1"
+	}
+	
+	public var blockNumber: UInt64 {
+		return "eip155:1"
+	}
```

Comment:
Could you provide me with what the default of these would be? @neekolas

Reply:
There is no default block number. I guess we can set it as 0 for EOAs.

There's two ways to go about getting the block number: we can make it the caller's problem and have them provide it at the time they are providing the signature (any eth client should be able to trivially look it up), or we can figure it out for them. Right now it's configured to happen the first way, but we might want to do it the second way because it's easier.

Reply:
The chainId is also UInt64? Do you want me to encrypt it before passing? Is that the expected format?

Reply:
No need to encrypt. It’s totally public